core/services/relay/evm/evm.go

```diff
@@ -75,14 +75,28 @@ func (r *Relayer) NewConfigProvider(args relaytypes.RelayArgs) (relaytypes.Confi
 
 type configWatcher struct {
 	utils.StartStopOnce
+	lggr             logger.Logger
 	contractAddress  common.Address
 	contractABI      abi.ABI
 	offchainDigester types.OffchainConfigDigester
 	configPoller     *ConfigPoller
 	chain            evm.Chain
+	new              bool
+	fromBlock        uint64
 }
 
 func (c *configWatcher) Start(ctx context.Context) error {
+	if c.new {
+		// Only replay if its a brand new job.
+		go func() {
+			c.lggr.Infow("starting replay for config", "fromBlock", c.fromBlock)
+			if err := c.configPoller.destChainLogPoller.Replay(context.Background(), int64(c.fromBlock)); err != nil {
+				c.lggr.Errorf("error replaying for config", "err", err)
+			} else {
+				c.lggr.Infow("completed replaying for config", "fromBlock", c.fromBlock)
+			}
+		}()
```

Comment:
This loose goroutine makes me nervous, even if it should be 'fast'. How about wiring up the usual stop channel and `WaitGroup` logic from `Close()`?

Reply:
added some handling